src/transformers/models/falcon_mamba/modeling_falcon_mamba.py

```diff
@@ -711,6 +711,11 @@ def prepare_inputs_for_generation(
         # Overwritten -- uses `cache_params` as opposed to `past_key_values`
 
         if use_cache:
+            cache_params_not_initialized = cache_params is None
+            if cache_params_not_initialized:
+                max_batch_size = inputs_embeds.size(0) if inputs_embeds is not None else input_ids.size(0)
+                cache_params = MambaCache(self.backbone.config, max_batch_size, device=self.device, dtype=self.dtype)
+                cache_position = torch.arange(0, self.backbone.config.conv_kernel, device=input_ids.device)
```

Comment:
why can't generate properly initialize this? (get_text_config()) could return backbone config or something no?